bundles/org.openhab.core.config.discovery.addon/src/main/java/org/openhab/core/config/discovery/addon/finders/UpnpAddonSuggestionFinder.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.config.discovery.addon.finders;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.jupnp.UpnpService;
+import org.jupnp.model.meta.RemoteDevice;
+import org.openhab.core.addon.AddonDiscoveryMethod;
+import org.openhab.core.addon.AddonInfo;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * This is a {@link UpnpAddonSuggestionFinder} for finding suggested Addons via UPnP.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = AddonSuggestionFinder.class, name = UpnpAddonSuggestionFinder.SERVICE_NAME)
+public class UpnpAddonSuggestionFinder extends BaseAddonSuggestionFinder {
```

Comment:
Same comment as for mDNS. This will not detect something if it is not yet in the UPnP registry. Devicesbeing added/removed in the network will not become visible.

Reply:
I think you can do it with UPnP as well, using the registerParticipant method of the UpnpIOService.

Reply:
This is how it looks in the UpnpDiscoveryService:

```
    @Override
    protected void startBackgroundDiscovery() {
        upnpService.getRegistry().addListener(this);
    }
```

Reply:
Yes. I  literally just found it myself by examining the source code. I should have read your response first. :)

Reply:
Done.